gitlint/utils.py

```diff
@@ -72,29 +73,45 @@ def _open_for_write(filename):
     return io.open(filename, 'w')
 
 
-def _get_cache_filename(name, filename):
-    """Returns the cache location for filename and linter name."""
+def _get_cache_filename(linter_hash, filename):
+    """Returns the cache location for filename and linter hash."""
     filename = os.path.abspath(filename)[1:]
     home_folder = os.path.expanduser('~')
     base_cache_dir = os.path.join(home_folder, '.git-lint', 'cache')
 
-    return os.path.join(base_cache_dir, name, filename)
+    return os.path.join(base_cache_dir, linter_hash, filename)
 
 
-def get_output_from_cache(name, filename):
+def calculate_hash(program, arguments):
+    """Calculate sha256 hash as hex string over program and arguments.
+
+    Args:
+      program: string: lint program.
+      arguments: list[string]: extra arguments for the program.
+
+    Returns: string: a string with the calculated sha256 hex value.
+    """
+    algorithm = hashlib.sha256()
+    algorithm.update(program)
```

Comment:
You also need to add a separator or otherwise:
`foo a b` would have the same has as `foo ab`.

Please add some tests as well.

Reply:
I changed the naming of the directory from using only the hash value to reusing the name and appending the hash separated by a dot. So it is now rather easy to see which directory contains to which linter and when something changes.

I've added the separator. I had to update the tests, which should test the new schema, too.